From 1738bae6a560fcfa56cf4a03a6b2f8f152189cb1 Mon Sep 17 00:00:00 2001
From: Doclic <doclicwastaken@gmail.com>
Date: Thu, 21 Jul 2022 04:44:33 +0200
Subject: [PATCH] Add PortalCooldown support


diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index ef96a15d..9f9bef42 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -376,6 +376,20 @@ public interface Entity extends Metadatable, CommandSender {
      * @param flag if the entity is silent
      */
     public void setSilent(boolean flag);
+
+    /**
+     * Gets the period of time (in ticks) before this entity can use a portal.
+     *
+     * @return portal cooldown ticks
+     */
+    int getPortalCooldown();
+
+    /**
+     * Sets the period of time (in ticks) before this entity can use a portal.
+     *
+     * @param cooldown portal cooldown ticks
+     */
+    void setPortalCooldown(int cooldown);
     // LumiereSpigot End
 
     // Spigot Start
-- 
2.34.1

